Added user defined context for Pdb

JamshedVesuna opened the pull request:
Closes #9087

JamshedVesuna committed:
Added user defined context for Pdb

Carreau commented:
You do not need to amend, you can just push more commit, that would have been helpful to see the difference between our PRs. 

The removal of `breaklist = self.get_file_breaks(filename)` was on purpose, it was unused, but agreed it is an unrelated change.

> Just created a new PR #9097 that includes your changes + some style fixes to stay consistent with the rest of the package.

The `if context is None` vs `if not context` is not a style issue, it behave differently for various values like `0` and `''` (empty string), that here make no sens (you at least want 1 line of context). But if we go this route, we might want to check for purely positive non null integers.

JamshedVesuna commented:
Ah apologies for the PR confusion. I've updated this PR to include your suggested changes. I included the removal of `breaklist = self.get_file_breaks(filename)` as I removed a few miscellaneous print statements as well.

I'd be happy to change it back to `if not context`, as it is cleaner (though that doesn't account for weird inputs like negative ints). 

I noticed that the debugger doesn't raise any exceptions, but rather prints them to stdout. However, when following this format, the error message is printed at the top of the stack trace, making it difficult to find, unless we `break` after printing. Let me know what works best.

Carreau commented:
I think this is fine let's move forward. We can refine later.
